### ISSUE TITLE
Let user choose padding

### ISSUE THREAD

**Issue by natir** [COMBINE-lab/kmers#7]
Hi,

Actually library, store kmer in an array of `u64` it could be nice to let user choose the type of array. I work with very short kmer, use and u64 to store it, would have a significant cost.

**Comment by Daniel-Liu-c0deb0t** [comment]
Rust-debruijn has an implementation with integers of various sizes: https://github.com/10XGenomics/rust-debruijn/blob/master/src/kmer.rs

**Comment by rob-p** [comment]
I like this idea, though I don't have a good sense of how much it will complexify the code.  In particular, will we have to have many specialize implementations to make use of word-size operations when the backing array is using word-sized elements, with fallbacks to slower implementations for smaller types, etc.?

**Comment by pmarks** [comment]
@rob-p yes it does make some code paths a bit more complicated. See the impls of [VarIntKmer](https://github.com/10XGenomics/rust-debruijn/blob/master/src/kmer.rs#L560) which supports kmers that are smaller than the bitwidth of the integer, vs [IntKmer](https://github.com/10XGenomics/rust-debruijn/blob/master/src/kmer.rs#L294) which requires that the kmer fills the integer.  There's usually a handful of extra shifts/mask to maintain the invariants of `VarIntKmer`, but it's not too crazy. There'd also be some overhead to deal with an array of smaller ints rather than a single in an array-backed kmer impl.
